### DISCUSSION TITLE
How to import realms in kc-quarkus built from source?

### DISCUSSION THREAD

**Discussion by mostermann96** [keycloak/keycloak#13166] [Q&A]
Hi,

I'm looking to do a proof-of-concept and need to build Keycloak from source.

However, I'm not sure, how to import realms / where the data/import directories should be, when building the quarkus distribution.
When using `java -jar server/target/lib/quarkus-run.jar start-dev --import-realm` all I get is a nullpointer exception.
`Could not invoke public void org.keycloak.quarkus.runtime.cli.command.ImportRealmMixin.setImportRealm(java.lang.String) with  (java.lang.NullPointerException)
`

**Comment by shobull** [comment]
I put exported `.json` realm file to `{KEYCLOAK_DIR}/data/import/exported-realm.json` and it works.

**Comment by buti1021** [comment]
I received that error when the volume I was mounting to was the json file instead of a directory including the json file
so use:
/keycloak-docker/import:/opt/keycloak/data/import
instead of:
/keycloak-docker/import/import.json:/opt/keycloak/data/import

However, the error message could be way better
